capirca/lib/paloaltofw.py

```diff
@@ -750,10 +752,27 @@ def __str__(self):
     rulebase = etree.SubElement(vsys_entry, "rulebase")
     security = etree.SubElement(rulebase, "security")
     rules = etree.SubElement(security, "rules")
+    tag = etree.Element("tag")
+
+    tag_num = 0
 
     # pytype: disable=key-error
     # pylint: disable=unused-variable
     for (header, pa_rules, filter_options) in self.pafw_policies:
+      tag_name = None
+      if header.comment:
+        comment = " ".join(header.comment).strip()
+        if comment:
+          tag_num += 1
+          # max tag len 127, max zone len 31
+          tag_name = self._TAG_NAME_FORMAT.format(
+            from_zone=filter_options[1], to_zone=filter_options[3],
+            num=tag_num)
+          tag_entry = etree.SubElement(tag, "entry",
+                                       {"name": tag_name})
+          comments = etree.SubElement(tag_entry, "comments")
+          comments.text = comment[:self._MAX_TAG_COMMENTS_LENGTH]
```

Comment:
This will silently elide the extra comment characters when the comment is too long. We should either error or, ideally, split the comment into multiple comments.

Reply:
I think truncation here is fine. The logic to create a new tag comment and associate multiple tags creates complicated, messy logic. This follows the expected behavior of adhering to comment limits, this shouldn't error out.

Reply:
policy description can also be truncated.